### DISCUSSION TITLE
Bare Metal Install Guide / Best Practice Config

### DISCUSSION THREAD

**Discussion by nsauter** [authelia/authelia#7202] [Q&A]
Hello!

I want to try out authelia but since i really dont like docker stuff on my homeservers i would like to install it on a bare metal / bare vm without docker. But unfortunately i cant find any docs or guides which help me setting everything up. Can anybody provide me some explanations and/or "default configs" to start with?

If desired i would be happy to contribute a guide to this repo as pull request to help new users in the future.

Thanks!

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Hello!

For deploying Authelia on a bare-metal or VM setup without using Docker, you can follow the guidelines provided in the Authelia documentation for bare-metal deployments. Here are the key steps and resources to help you get started:

1. **Systemd**: Authelia can be run as a daemon using systemd. The documentation provides example systemd unit files (`authelia.service` and `authelia@.service`) which you can use to set up Authelia to start automatically on system boot.

2. **Installation**:
   - **Arch Linux**: Authelia is available as an AUR package which you can install using an AUR helper like `yay` or `pacman`.
   - **Debian and Debian-based systems**: You can install Authelia using the provided `.deb` packages. Additionally, Authelia maintains an APT repository which you can use to keep the software up-to-date easily.
   - **Nix**: Using the Nix package manager, Authelia can be installed from the `nixpkgs-unstable` channel.
   - **FreeBSD**: Authelia is available through FreshPorts, and an `rc.d` script (`authelia-fb-rc.d`) is provided for service management.
   - **Binaries**: Directly download and install the binaries from the Authelia GitHub releases page suitable for your operating system.

3. **Configuration**: After installation, you will need to configure Authelia. The configuration typically involves setting up the authentication backend, session storage, and defining the access control rules. Authelia uses a YAML configuration file for these settings.

4. **Get Started Guide**: It is strongly recommended to read the [Get Started Guide](https://www.authelia.com/integration/deployment/introduction) provided by Authelia, which walks you through the essential steps for bootstrapping Authelia.

5. **Contribution**: If you create a guide based on your experience, it would be a valuable contribution to the community. You can follow the contribution guidelines provided by Authelia to submit your guide as a pull request.

For detailed instructions and default configuration examples, please refer to the specific sections of the Authelia documentation related to bare-metal deployment.

If you need further assistance or specific configuration examples, feel free to ask!

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by hendrik1120** [comment]
Did you already take a look at https://www.authelia.com/integration/deployment/bare-metal/?

**Reply by nsauter** [reply]
Yes i did. But to be honest: Thats far away from a good explanation what to do. Because of this i was asking and offering to write a good how to start guide.

**Reply by james-d-elliott** [reply]
If you'd like to contribute to the system agnostic bare metal guide then we'd welcome it. We're not interested in maintaining guides for individual bare-metal flavors.

**Reply by mondij** [reply]
I wanna try it too but I can't find the exact steps either.

Can someone please provide exact, step by step, Linux commands to run. Let's say Ubuntu.

I looked at https://www.authelia.com/integration/deployment/bare-metal/ but it doesn't make sense to me either. 
I don't trust chatgpt with it's hallucinations.

**Reply by hendrik1120** [reply]
Since Ubuntu is based on debian you can install the debian package from https://apt.authelia.com/stable/debian/
You can follow either the instructions on that page which are a bit outdated or any other instructions on how to install .deb packages.

```shell
sudo apt install ca-certificates curl
sudo curl -fsSL https://apt.authelia.com/organization/signing.asc -o /usr/share/keyrings/authelia.asc
echo \
  "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/authelia.asc] https://apt.authelia.com/stable/debian/debian all main" | \
  sudo tee /etc/apt/sources.list.d/authelia.list > /dev/null
```

After that you can just `apt install authelia` and run the `authelia` command which will generate a `configuration.yml`. That you will need to configure using the docs.
If you want to use systemd to manage authelia, there are many other guides out there.

**Comment by idc77** [comment]
He's right though. The https://www.authelia.com/integration/prologue/get-started/ guide is overly confusing.

**Reply by james-d-elliott** [reply]
This is not really the same topic, none the less we'd welcome a contribution.
